### PR TITLE
Replace config template by conversion to yaml

### PR DESCRIPTION
**Pull request by lbetz**

### CODE REVIEW

File: manifests/config.pp [hunk 1]
```diff
@@ -6,11 +6,20 @@
 class icingadb::config {
   assert_private()
 
+  $stdlib_version  = $icingadb::globals::stdlib_version
   $conf_dir        = $icingadb::globals::conf_dir
   $user            = $icingadb::globals::user
   $group           = $icingadb::globals::group
   $redis_tls_files = $icingadb::redis_tls_files
   $db_tls_files    = $icingadb::db_tls_files
+  $config_file     = inline_epp('<%= $preamble %><%= $content -%>', {
```
Comment (rwaffen) [inline manifests/config.pp:15]: Don't do this please....

Comment (lbetz) [reply manifests/config.pp:15]: Only the inline is the problem or a template file is also a no go?

File: manifests/config.pp [hunk 2]
```diff
@@ -6,11 +6,17 @@
...
   $conf_dir        = $icingadb::globals::conf_dir
   $user            = $icingadb::globals::user
   $group           = $icingadb::globals::group
   $redis_tls_files = $icingadb::redis_tls_files
   $db_tls_files    = $icingadb::db_tls_files
+  $config_file     = if versioncmp($stdlib_version, '9.0.0') < 0 {
+    to_yaml($icingadb::config)
+  } else {
+    stdlib::to_yaml($icingadb::config)
+  }
```
Comment (ekohl) [inline manifests/config.pp:19]: There was talk of introducing a stdlib 8.7.0 with the missing deprecated functions, but I don't think anything happened on that front.

I wonder: can you use `if defined("stdlib::to_yaml')` to dynamically detect it?

File: manifests/config.pp [hunk 3]
```diff
@@ -6,11 +6,17 @@
 class icingadb::config {
   assert_private()
 
+  $stdlib_version  = $icingadb::globals::stdlib_version
   $conf_dir        = $icingadb::globals::conf_dir
   $user            = $icingadb::globals::user
   $group           = $icingadb::globals::group
   $redis_tls_files = $icingadb::redis_tls_files
   $db_tls_files    = $icingadb::db_tls_files
+  $config_file     = if versioncmp($stdlib_version, '9.0.0') < 0 {
```
[the remark this answers is not in the document]
Comment (lbetz) [reply manifests/config.pp:15]: I don't care about the name, which one would be better?

File: manifests/config.pp (hunk 2 above)
Comment (lbetz) [reply manifests/config.pp:19]: If it should be that simple ...

Comment (lbetz) [reply manifests/config.pp:19]: ... unfortunately it is not.

File: manifests/config.pp (hunk 3 above)
[the remark this answers is not in the document]
Comment (lbetz) [reply manifests/config.pp]: Now I chose config_content.

File: manifests/config.pp [hunk 4]
```diff
@@ -6,11 +6,17 @@
...
   assert_private()
 
+  $stdlib_version  = $icingadb::globals::stdlib_version
   $conf_dir        = $icingadb::globals::conf_dir
   $user            = $icingadb::globals::user
   $group           = $icingadb::globals::group
   $redis_tls_files = $icingadb::redis_tls_files
   $db_tls_files    = $icingadb::db_tls_files
+  $config_content  = if versioncmp($stdlib_version, '9.0.0') < 0 {
+    to_yaml($icingadb::config)
```
Comment (bastelfreak) [inline manifests/config.pp:16]: are there still users for this old version of stdlib? metadata.json lists 9.0.0 als oldest supported version.

Comment (lbetz) [reply manifests/config.pp:16]: Yes, I have a customer this week who is not on the bleeding edge.

Comment (lbetz) [reply manifests/config.pp:16]: That's the reason for the other three Icinga modules to also support older version.

Comment (ekohl) [reply manifests/config.pp:16]: If that version is supported, `metadata.json` should reflect that

Comment (lbetz) [reply manifests/config.pp:16]: It does?!
https://github.com/voxpupuli/puppet-icingadb/blob/54fa5822604f95978ac978e7bd0ce50711aa2526/metadata.json#L12

Comment (ekohl) [reply manifests/config.pp:16]: Sorry, I went @bastelfreak's comment

**Review by ekohl — APPROVED** [review]
I'm still annoyed that stdlib didn't have an 8.x version with the namespaced functions and we have to resort to this

File: manifests/config.pp (hunk 4 above)
Comment (lbetz) [reply manifests/config.pp:16]: @ekohl You're welcome.

**Pull request merged**